tests/30rooms/12thirdpartyinvite.pl

```diff
@@ -240,24 +230,30 @@ sub do_3pid_invite {
    )
 }
 
-sub start_id_server
+sub id_server_fixture
 {
-   my $id_server = SyTest::Identity::Server->new;
-   $loop->add( $id_server );
-   $id_server->listen(
-      host    => "localhost",
-      service => "",
-      extensions => [qw( SSL )],
-      # Synapse currently only talks IPv4
-      family => "inet",
-
-      SSL_cert_file => "$DIR/../../keys/tls-selfsigned.crt",
-      SSL_key_file => "$DIR/../../keys/tls-selfsigned.key",
-   )->then( sub {
-      my ( $listener ) = @_;
-
-      my $sock = $listener->read_handle;
-      my $id_server_hostandport = sprintf "%s:%d", $sock->sockhostname, $sock->sockport;
-      Future->done( $id_server );
-   })
+   return fixture(
+      setup => sub {
+         my $id_server = SyTest::Identity::Server->new;
+         $loop->add( $id_server );
+
+         $id_server->listen(
+            host    => "localhost",
+            service => "",
+            extensions => [qw( SSL )],
+            # Synapse currently only talks IPv4
+            family => "inet",
+
+            SSL_cert_file => "$DIR/../../keys/tls-selfsigned.crt",
+            SSL_key_file => "$DIR/../../keys/tls-selfsigned.key",
+         )->then_done( $id_server );
+      },
+
+      teardown => sub {
+         my ( $id_server ) = @_;
+         $loop->remove( $id_server );
```

Comment:
This should also kill any existing connections, in case a homeserver is keep-alive-ing